flutter_local_notifications/windows/flutter_local_notifications_plugin.cpp

```diff
@@ -0,0 +1,308 @@
+		channel(channel) {}
+
+	FlutterLocalNotificationsPlugin::~FlutterLocalNotificationsPlugin() {}
+
+	void FlutterLocalNotificationsPlugin::HandleMethodCall(
+		const flutter::MethodCall<flutter::EncodableValue>& method_call,
+		std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result
+	) {
+		const auto& method_name = method_call.method_name();
+		if (method_name == Method::GET_NOTIFICATION_APP_LAUNCH_DETAILS) {
+			result->Success();
+		}
+		else if (method_name == Method::INITIALIZE) {
+			const auto args = std::get_if<flutter::EncodableMap>(method_call.arguments());
+			if (args != nullptr) {
+				const auto appName = Utils::GetMapValue<std::string>("appName", args).value();
+				const auto aumid = Utils::GetMapValue<std::string>("aumid", args).value();
+				const auto iconPath = Utils::GetMapValue<std::string>("iconPath", args);
+				const auto iconBgColor = Utils::GetMapValue<std::string>("iconBgColor", args);
+
+				result->Success(Initialize(appName, aumid, iconPath, iconBgColor));
+			}
+			else {
+				result->Error("INTERNAL", "flutter_local_notifications encountered an internal error.");
+			}
+		}
+		else if (method_name == Method::SHOW) {
+			channel->InvokeMethod("test", nullptr);
+
+			const auto args = std::get_if<flutter::EncodableMap>(method_call.arguments());
+			if (args != nullptr && toastNotifier.has_value()) {
+				const auto id = Utils::GetMapValue<int>("id", args).value();
+				const auto title = Utils::GetMapValue<std::string>("title", args);
+				const auto body = Utils::GetMapValue<std::string>("body", args);
+				const auto payload = Utils::GetMapValue<std::string>("payload", args);
+				const auto group = Utils::GetMapValue<std::string>("group", args);
+
+				ShowNotification(id, title, body, payload, group);
+				result->Success();
+			}
+			else {
+				result->Error("INTERNAL", "flutter_local_notifications encountered an internal error.");
+			}
+		}
+		else if (method_name == Method::CANCEL && toastNotifier.has_value()) {
+			const auto args = std::get_if<flutter::EncodableMap>(method_call.arguments());
+			if (args != nullptr) {
+				const auto id = Utils::GetMapValue<int>("id", args).value();
+				const auto group = Utils::GetMapValue<std::string>("group", args);
+
+				CancelNotification(id, group);
+				result->Success();
+			}
+			else {
+				result->Error("INTERNAL", "flutter_local_notifications encountered an internal error.");
+			}
+		}
+		else if (method_name == Method::CANCEL_ALL && toastNotifier.has_value()) {
+			CancelAllNotifications();
+			result->Success();
+		}
+		else {
+			result->NotImplemented();
+		}
+	}
+
+	std::optional<bool> FlutterLocalNotificationsPlugin::HasIdentity() {
+		if (!IsWindows8OrGreater()) {
+			// OS is windows 7 or lower
+			return false;
+		}
+
+		UINT32 length;
+		auto err = GetCurrentPackageFullName(&length, nullptr);
+		if (err != ERROR_INSUFFICIENT_BUFFER) {
```

Comment:
When I run the example here, it returns the error code 87(ERROR_INVALID_PARAMETER) which causes the initialization to fail. But when I still return false(not std:nullopt) here, the plugin still works fine.

Reply:
I found the cause of the problem. The variable `length` needs to be initialized.
Replace `UINT32 length;` to `UINT32 length = 0;` could resolve this issue.

Reply:
Wow! Can't wait! Thank you guys for working on this.